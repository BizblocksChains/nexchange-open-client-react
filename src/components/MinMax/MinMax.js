import React from 'react';
import { I18n } from 'react-i18next';
import isFiatOrder from 'Utils/isFiatOrder';
import styles from './MinMax.scss';

const MinMax = props => {
  if (props.order && isFiatOrder(props.order) && props.type === 'Deposit') return null;

  return (
    <I18n ns="translations">
      {t => (
        <div className={`${styles.minmax} ${props.home ? styles.home : ''}`}>
          <p className={props.min > props.amount && props.amount !== '...' ? 'error' : ''}>
            {t('exchangewidget.min')}: {props.min}
          </p>

          <p className={props.max < props.amount && props.amount !== '...' ? 'error' : ''}>
            {t('exchangewidget.max')}: {props.max}
          </p>
        </div>
      )}
    </I18n>
  );
};

export default MinMax;
