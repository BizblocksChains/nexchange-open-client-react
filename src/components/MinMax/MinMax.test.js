import React from 'react';
import { shallow } from 'enzyme';
import orderFiat from 'Mocks/orderFiat';
import order from 'Mocks/order';
import MinMax from './MinMax';

describe('MinMax', () => {
  it('renders correctly correct amounts', () => {
    const wrapShallow = shallow(<MinMax min={1} max={2} amount={1.5} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={1.5} order={order} type="Deposit" />).dive();
    const wrapShallowFiatDeposit = shallow(<MinMax min={1} max={2} amount={1.5} order={orderFiat} type="Deposit" />);
    const wrapShallowFiatReceive = shallow(<MinMax min={1} max={2} amount={1.5} order={orderFiat} type="Receive" />).dive();

    expect(wrapShallow).toMatchSnapshot();
    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowFiatDeposit).toMatchSnapshot();
    expect(wrapShallowFiatReceive).toMatchSnapshot();
  });

  it('renders correctly too low amounts', () => {
    const wrapShallow = shallow(<MinMax min={1} max={2} amount={0.5} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={0.5} order={order} type="Deposit" />).dive();
    const wrapShallowFiat = shallow(<MinMax min={1} max={2} amount={0.5} order={orderFiat} type="Deposit" />).dive();

    expect(wrapShallow).toMatchSnapshot();
    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowFiat).toMatchSnapshot();
  });

  it('renders correctly too high amounts', () => {
    const wrapShallow = shallow(<MinMax min={1} max={2} amount={3} home="true" order={order} type="Deposit" />).dive();
    const wrapShallowHome = shallow(<MinMax min={1} max={2} amount={3} order={order} type="Deposit" />).dive();
    const wrapShallowFiat = shallow(<MinMax min={1} max={2} amount={3} order={orderFiat} type="Deposit" />).dive();

    expect(wrapShallow).toMatchSnapshot();
    expect(wrapShallowHome).toMatchSnapshot();
    expect(wrapShallowFiat).toMatchSnapshot();
  });
});
