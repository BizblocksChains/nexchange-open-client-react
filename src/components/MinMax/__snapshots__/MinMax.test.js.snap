// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinMax renders correctly correct amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly correct amounts 4`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too high amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className="error"
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className="error"
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too high amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className="error"
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className="error"
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too high amounts 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className=""
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className="error"
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className=""
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className="error"
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too low amounts 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="error"
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="error"
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too low amounts 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="error"
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="error"
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MinMax renders correctly too low amounts 3`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <I18n
    ns="translations"
>
    [Function]
</I18n>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="error"
>
          exchangewidget.min
          : 
          1
</p>,
        <p
          className=""
>
          exchangewidget.max
          : 
          2
</p>,
      ],
      "className": "minmax ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "className": "error",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.min",
          ": ",
          1,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "className": "",
        },
        "ref": null,
        "rendered": Array [
          "exchangewidget.max",
          ": ",
          2,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="error"
>
            exchangewidget.min
            : 
            1
</p>,
          <p
            className=""
>
            exchangewidget.max
            : 
            2
</p>,
        ],
        "className": "minmax ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.min",
              ": ",
              1,
            ],
            "className": "error",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.min",
            ": ",
            1,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "exchangewidget.max",
              ": ",
              2,
            ],
            "className": "",
          },
          "ref": null,
          "rendered": Array [
            "exchangewidget.max",
            ": ",
            2,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
